### PR TITLE
add evidently monitoring

### PR DESCRIPTION
**Pull request by ajshedivy**
Build and test components that "containerize" Evidently functionality.

### Overview
Evidently is an open source library that helps evaluate and test data and ML quality throughout model lifecycle. Evidently offers three main components: Reports, Tests, and Monitors.

### Current changes
- `DataQualityPreset` component
- `html-viewer` base component for outputting static html in kf pipeline UI

### TODO
- [x] create base Evidently preset component
- [x] clean up testing framework for new components
- [x] build example pipeline with Evidently components
- [x] code clean up/ formatting

Evidently documentation: https://docs.evidentlyai.com/

### CODE REVIEW

**Comment by ajshedivy** [comment]
### Next changes
- formalize Evidently component input/output parameters (what is best practice/most common)
- configure `kfp.client()` for testing support

File: monitoring/evidently/data-quality-report/data-quality-report-component.ipynb [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "BASE_IMAGE = \"quay.io/ibm/kubeflow-notebook-image-ppc64le:latest\"\n",
+    "\n",
+    "def data_quality_report(\n",
+    "    cur_path: InputPath('CSV'),\n",
```
Comment (lehrig) [inline monitoring/evidently/data-quality-report/data-quality-report-component.ipynb]: 1. rename: cur_path -> dataset_path
2. It's good to support CSV but we should also support DFs directly. I propose to add another parameter "dataset_type", which is either "DataFrame", "CSV", or "HuggingFace". I'd default to DF. Obviously, the code needs some adaptation to handle that.

File: monitoring/evidently/data-quality-report/data-quality-report-component.ipynb [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    "    output_path: OutputPath('HTML'),\n",
+    ") -> None:\n",
+    "\n",
+    "    import pandas as pd\n",
```
Comment (lehrig) [inline monitoring/evidently/data-quality-report/data-quality-report-component.ipynb:52]: Add documentation to describe this methods (this is helpful for users who want to use this component; some Kubeflow tools like Elyra display this documentation). Use this as template: https://github.com/lehrig/kubeflow-ppc64le-components/blob/main/model-building/monitor-training/monitor-training.ipynb

File: monitoring/evidently/data-quality-report/data-quality-report-component.ipynb [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    "    \n",
+    "data_quality_report_op = create_component_from_func(\n",
+    "    data_quality_report,\n",
+    "    output_component_file='component.yaml',\n",
+    "    base_image=BASE_IMAGE,\n",
+    "    packages_to_install=[\n",
+    "        'evidently==0.2.0',\n",
+    "    ],\n",
+    "    annotations={\n",
+    "        'author': 'Evidently AI'\n",
```
Comment (lehrig) [inline monitoring/evidently/data-quality-report/data-quality-report-component.ipynb]: I think this annotation can be removed

File: monitoring/evidently/data-quality-report/data-quality-report-component.ipynb [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    "    df = pd.read_csv(cur_path)\n",
+    "\n",
+    "    report = Report(metrics=[\n",
+    "        DataQualityPreset()\n",
+    "    ])\n",
+    "    \n",
+    "    report.run(current_data=df, reference_data=None, column_mapping=None)\n",
+    "    \n",
+    "    Path(output_path).parent.mkdir(parents=True, exist_ok=True)\n",
+    "    report.save_html(output_path)\n",
```
Comment (lehrig) [inline monitoring/evidently/data-quality-report/data-quality-report-component.ipynb]: Additionally render output as Visualization (as the HTML viewer would do as well)

File: monitoring/evidently/data-quality-report/test-data-quality-report.ipynb [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "DATA_QUALITY = os.path.abspath(\"\")\n",
+    "EVIDENTLY    = os.path.dirname(os.path.dirname(DATA_QUALITY))\n",
+    "HTML_VIWER   = os.path.join(EVIDENTLY, 'html-viewer')\n",
```
Comment (lehrig) [inline monitoring/evidently/data-quality-report/test-data-quality-report.ipynb]: Typo: HTML_VIWER -> HTML_VIEWER

File: monitoring/html-viewer/html-viewer-component.ipynb [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    "from kfp.components import InputPath, OutputPath, create_component_from_func\n",
+    "\n",
+    "BASE_IMAGE = \"quay.io/ibm/kubeflow-notebook-image-ppc64le:latest\"\n",
+    "\n",
+    "def view_html(\n",
+    "    html_path: InputPath('HTML'),\n",
+    "    mlpipeline_ui_metadata_path: OutputPath(),\n",
+    ") -> None:\n",
+    "\n",
+    "    import os\n",
```
Comment (lehrig) [inline monitoring/html-viewer/html-viewer-component.ipynb:18]: Add documentation (see above)

File: monitoring/html-viewer/html-viewer-component.ipynb [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+    "    \n",
+    "    \n",
+    "\n",
+    "\n",
+    "data_quality_report_op = create_component_from_func(\n",
+    "    view_html,\n",
+    "    output_component_file='component.yaml',\n",
+    "    base_image=BASE_IMAGE,\n",
+    "    annotations={\n",
+    "        'author': 'Evidently AI'\n",
```
Comment (lehrig) [inline monitoring/html-viewer/html-viewer-component.ipynb:43]: remove annotation

File: monitoring/html-viewer/html-viewer-component.ipynb [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+    "    \n",
+    "    metadata = {\n",
+    "        'outputs' : [{\n",
+    "        'type': 'web-app',\n",
+    "        'storage': 'inline',\n",
+    "        'source': html_content,\n",
+    "        }, {\n",
+    "        'type': 'web-app',\n",
+    "        'storage': 'inline',\n",
+    "        'source': '<h1>Hello, World!</h1>',\n",
```
Comment (lehrig) [inline monitoring/html-viewer/html-viewer-component.ipynb]: Remove second output (Hello World wep-app)

**Review by lehrig — CHANGES_REQUESTED** [review]
Nice work! I mainly had minor comments + don't forget removing .DS_Store.

Regarding the expansion to DF/CSV/Huggingface Datasets: Ping me if you need help :)

File: monitoring/evidently/data-quality-report/data-quality-report-component.ipynb (hunk 1 above)
Comment (ajshedivy) [reply monitoring/evidently/data-quality-report/data-quality-report-component.ipynb]: I added support for dataframe and csv. the dataframe serialization method im using is pickle. We should discuss the best practices for passing dataframes between components. I see that some examples use `np.savez()` and some use pickle or parquet.

File: monitoring/evidently/data-quality-report/data-quality-report-component.ipynb (hunk 2 above)
Comment (ajshedivy) [reply monitoring/evidently/data-quality-report/data-quality-report-component.ipynb:52]: forthcoming ...

**Pull request merged**